command.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2016 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package updater
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/keybase/go-logging"
+)
+
+func timerWithTimeout(timeout time.Duration, cmd *exec.Cmd, log logging.Logger) *time.Timer {
+	return time.AfterFunc(timeout, func() {
+		if cmd != nil && cmd.Process != nil {
+			log.Warningf("Command timed out, killing")
+			if err := cmd.Process.Kill(); err != nil {
+				log.Warningf("Error trying to kill process: %s", err)
+			}
+		}
+	})
```

Comment:
Calling Output or CombinedOutput, starts, waits and gets output all in a single call. I think I prefer using a timer than doing Start() and then Wait() with channel code?

As far as being graceful...

> Everything will probably be just fine if you kill-9 something. If a program fails permanently and dramatically when it's kill-9'd, you should "remove it from your filesystem", because it can't handle other, unavoidable abrupt failures either.

Also there are problems with your kill code... you are assuming the process object is always there and if the process dies after TERM timeout it will error with Process.Kill() even though it got killed.